server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -140,27 +142,50 @@ private MappedFieldType getDynamicField(String field) {
     }
 
     /**
-     * Returns all the mapped field types.
+     * Returns all the mapped field types that match a pattern
+     *
+     * Note that if a field is aliased and both its actual name and its alias
+     * match the pattern, the returned collection will contain the field type
+     * twice.
      */
-    Collection<MappedFieldType> get() {
-        return fullNameToFieldType.values();
+    Collection<MappedFieldType> getMatchingFieldTypes(String pattern) {
```

Comment:
I think it'd be nice to keep a convenience method like `getAllFieldTypes`. It doesn't really add complexity and makes the call sites easier to understand.

Reply:
Really we want to remove `get()` as it's only used by the parent-join module to keep track of a singleton field type, so this felt like a convenient halfway house.  I can put it back in if you prefer though.

Reply:
I don't feel strongly, but to me it helps. I also see it used in `MapperService`, and we might be able to replace some instances of `getMatchingFieldNames("*")` in tests?